### ISSUE TITLE
EBS volume creation does not honor aws:RequestTag

### ISSUE THREAD

**Issue by uasalian** [terraform-providers/terraform-provider-aws#6179]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.8
+ provider.aws v1.37.0
```

### Affected Resource(s)

```
aws_ebs_volume
```

### Terraform Configuration Files

```hcl
resource "aws_ebs_volume" "myVolume" {
  availability_zone = "us-east-1a"
  size              = "80"
  type              = "gp2"

  tags = {
    Name            = "alpha-Test"
  }
}
```

### Expected Behavior

Should create EBS volume

### Actual Behavior

```
Error: Error applying plan:

1 error(s) occurred:

* aws_ebs_volume.my_volume: 1 error(s) occurred:

* aws_ebs_volume.my_volume: Error creating EC2 volume: UnauthorizedOperation: You are not authorized to perform this operation. Encoded authorization failure message:
```

### Steps to Reproduce

1. Attach AWS managed policy `PowerUserAccess` to the IAM user/role executing terraform
2. Also add the following policy (set the right account id):
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Deny",
      "Action": [
        "ec2:CreateVolume",
        "ec2:RunInstances"
      ],
      "Resource": [
          "arn:aws:ec2:us-east-1:123456789012:instance/*",
          "arn:aws:ec2:us-east-1:123456789012:volume/*"
      ],
      "Condition": {
         "StringNotLike": {
             "aws:RequestTag/Name": "alpha-*"
          }
       }
    }
  ]
}
```
3. `terraform apply`

### Important Factoids

The idea is to enforce setting a tag in the form `Name=alpha-*` while creating EC2 instance and EBS volume. The policy can be tested in AWS CLI (or Console) using:
```
aws  ec2 create-volume --availability-zone us-east-1a --volume-type gp2 --size 80 \
 --tag-specifications 'ResourceType=volume,Tags=[{Key=Name,Value=alpha-Test}]'
```
But, terraform errors out while creating EBS volume.

**Comment by wedge-jarrad** [comment]
This looks like a duplicate of #1533 and #4428 which were fixed by #6396.

**Comment by bflad** [comment]
I believe @wedge-jarrad is correct above, closing this issue as it should have been resolved in version 1.44.0 of the Terraform AWS Provider.
